walkdir.py

```diff
@@ -202,11 +202,32 @@ def onloop(dirpath):
                     continue
         yield dir_entry
 
+
+def handle_dirsymlink_as_file(walk_iter):
+    """Handle symlink to directories as files during the walk.
+
+       By default os.walk insert symlink to directories to the subdirs list.
+       This function move symlink to directories from the subdirs list to the file list.
+
+       NOTE: This function is not needed when using the followlinks option.
+       when following links the dirsymlinks will be walked when entering the linked dir.
+    """
+    for dir_entry in walk_iter:
+        dirpath = dir_entry[0]
+        subdirs = dir_entry[1]
+        files = dir_entry[2]
+        for subdir in subdirs:
+            if os.path.islink(os.path.join(dirpath, subdir)):
+                files.append(subdir)
+                subdirs.remove(subdir)
+        yield dir_entry
+
 # Convenience function that puts together an iterator pipeline
 
 def filtered_walk(top, included_files=None, included_dirs=None,
                        excluded_files=None, excluded_dirs=None,
-                       depth=None, followlinks=False, min_depth=None):
+                       depth=None, followlinks=False, min_depth=None,
+                       dirsymlink=False):
```

Comment:
I think `dirsymlinks_as_files` (to match the handler name) would be a more self-explanatory option name. (Just "dirsymlink" on its own doesn't convey enough info about the behaviour change)

The docstring also needs to be updated for the new option.